python/pyspark/sql/classic/dataframe.py

```diff
@@ -1850,6 +1850,12 @@ def toArrow(self) -> "pa.Table":
     def toPandas(self) -> "PandasDataFrameLike":
         return PandasConversionMixin.toPandas(self)
 
+    def transpose(self, indexColumn: Optional[Column] = None) -> ParentDataFrame:
```

Comment:
shall we support column name as well?

Reply:
Make sense, added [5c8c069](https://github.com/apache/spark/pull/47884/commits/5c8c069fcf0298292910e24586756dc47111110e).